src/main/java/com/ibm/restconf/service/impl/KafkaExternalMessagingServiceImpl.java

```diff
@@ -34,10 +35,13 @@ public KafkaExternalMessagingServiceImpl(RCDriverProperties properties, KafkaTem
         this.objectMapper = objectMapper;
     }
 
-    @Override public void sendExecutionAsyncResponse(ExecutionAsyncResponse request) {
+    @Override public void sendExecutionAsyncResponse(ExecutionAsyncResponse request, String tenantId) {
         try {
             final String message = objectMapper.writeValueAsString(request);
-            ListenableFuture<SendResult<String, String>> future = kafkaTemplate.send(properties.getTopics().getLifecycleResponsesTopic(), message);
+            logger.info("TenantId in Kafka==> "+ tenantId);
+            ProducerRecord<String, String> producerRecord = new ProducerRecord<>(properties.getTopics().getLifecycleResponsesTopic(), message);
+            producerRecord.headers().add("TenantId", tenantId.getBytes());
```

Comment:
This will give NullPonterException when tested with pre CP4NA2.4, where tenantId is not passed at all.

Reply:
Ok, I see that you are making default value to 1, if this null in the Controller class.